File: Numeros Pseudo-Aleatorios/IBM RANDU/randu.cs
/*
   PRNG .......: RANDU
   Ano ........: Decada de 1960
   Autor ......: IBM
*/

using System;

namespace PRNG
{
    class Program
    {
      
        static long RANDU(ref long seed)
        {
            seed = (65539 * seed) % 2147483648;
            return seed;
        }

        static void Main(string[] args)
        {
          
            long Xn1 = 1234;
            long Xn2 = DateTimeOffset.Now.ToUnixTimeSeconds();

            Console.WriteLine("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)");
            Console.WriteLine("PSEUDORANDOM NUMBER GENERATOR (PRNG)");
            Console.WriteLine("PADRAO: RANDU");
            Console.WriteLine("==========================================");
            Console.WriteLine();
            Console.WriteLine("    Semente: 1234        Semente: TIME");
            Console.WriteLine("    -------------        -------------");

            for (int i = 1; i <= 10; i++)
              Console.WriteLine($"{RANDU(ref Xn1),17}        {RANDU(ref Xn2),13}");
          
        }
    }
  
}

File: Numeros Pseudo-Aleatorios/IBM RANDU/randu.py
"""
  PRNG .......: RANDU
  Ano ........: Decada de 1960
  Autor ......: IBM
"""

import time

def RANDU(seed):
    return (65539 * seed) % 2147483648

def main():
    Xn1 = 1234
    Xn2 = int(time.time())

    print("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)")
    print("PSEUDORANDOM NUMBER GENERATOR (PRNG)")
    print("PADRAO: RANDU")
    print("==========================================")
    print()
    print("    Semente: 1234        Semente: TIME")
    print("    -------------        -------------")

    for i in range(1, 11):
        Xn1 = RANDU(Xn1)
        Xn2 = RANDU(Xn2)
        print(f"{Xn1:17}        {Xn2:13}")

if __name__ == "__main__":
    main()
